src/Surfnet/StepupGateway/GatewayBundle/Tests/Service/GatewayLoginServiceTest.php

```diff
@@ -169,18 +180,131 @@ public function it_should_on_sso_return_a_proxy_request()
+                </ds:Reference>
+            </ds:SignedInfo>
+            <ds:SignatureValue>QZr...==</ds:SignatureValue>
+            <ds:KeyInfo>
+                <ds:X509Data>
+                    <ds:X509Certificate>MII...=</ds:X509Certificate>
+                </ds:X509Data>
+            </ds:KeyInfo>
+        </ds:Signature>
+        <saml:Subject>
+            <saml:NameID Format="urn:oasis:names:tc:SAML:2.0:nameid-format:persistent">
+                724cca6778a1d3db16b65c40d4c378d011f220be
+            </saml:NameID>
+            <saml:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
+                <saml:SubjectConfirmationData
+                        IssueInstant="2014-10-22T11:09:59Z"
+                        Recipient="https://gateway.org/acs"
+                        InResponseTo="_mocked_generated_id"/>
+            </saml:SubjectConfirmation>
+        </saml:Subject>
+        <saml:Conditions NotBefore="2014-10-22T11:07:07Z" NotOnOrAfter="2014-10-22T11:12:08Z">
+            <saml:AudienceRestriction>
+                <saml:Audience>https://gateway.org/metadata</saml:Audience>
+            </saml:AudienceRestriction>
+        </saml:Conditions>
+        <saml:AuthnStatement
+                AuthnInstant="2014-10-22T11:07:07Z"
+                SessionNotOnOrAfter="2014-10-22T19:07:07Z"
+                SessionIndex="_1dad5d4bf289a5761a62fedf91143816d323a0604b">
+            <saml:AuthnContext>
+                <saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:Password</saml:AuthnContextClassRef>
+                <saml:AuthenticatingAuthority>https://proxied-idp.edu/</saml:AuthenticatingAuthority>
+            </saml:AuthnContext>
+        </saml:AuthnStatement>
+        <saml:AttributeStatement>
+            <saml:Attribute Name="urn:oid:0.9.2342.19200300.100.1.3"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
+                <saml:AttributeValue xsi:type="xs:string">john.doe@example.edu</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="urn:oid:1.3.6.1.4.1.5923.1.1.1.10"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
+                <saml:AttributeValue>
+                    <saml:NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified">
+                        urn:collab:person:example.edu:jdoe
+                    </saml:NameID>
+                </saml:AttributeValue>
+            </saml:Attribute>
+        </saml:AttributeStatement>
+    </saml:Assertion>
+</samlp:Response>
+';
+
+        $this->mockPostBinding($samlResponseXml);
+
+
+        $samlResponse = base64_encode($samlResponseXml);
 
         $httpRequest = Request::create($proxyRequest->getDestination(). '?' .$proxyRequest->buildRequestQuery());
         $httpRequest->request->set('SAMLResponse', $samlResponse);
 
+        // Consume assertion
```

Comment:
Maybe explain in more detail to the reader/maintainer how the test works, some additional inline doc wouldn't hurt.